File: frontend/bff/src/application/files/dtos/file-kind-summary.dto.ts
import type { FileKind } from '../../../cross-cutting/enums/file-kind.js';

export interface FileKindSummaryDto {
  kind: FileKind;
  total: number;
  processing: number;
  withErrors: number;
  lastUploadedAt: string | null;
  /** O usuário tem a regra exigida para enviar este tipo. */
  canUpload: boolean;
}

File: backend/storage-service/src/Fix.Storage.Application/UseCases/Files/Dtos/FileKindSummaryDto.cs
using Fix.Storage.Domain.AggregateRoots.Files;

namespace Fix.Storage.Application.Files.Dtos;

public sealed record FileKindSummaryDto(
    FileKind Kind,
    int Total,
    int Processing,
    int WithErrors,
    DateTimeOffset? LastUploadedAt,
    // O usuário tem a regra exigida pelo tipo.
    bool CanUpload);
